.github/workflows/codeql-analysis.yml

```
@@ -0,0 +1,32 @@
+name: "CodeQL"
+# see https://codeql.github.com/docs/
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+
+jobs:
+  analyse:
+    name: Analyse
+    runs-on: ubuntu-latest
+
```

Please add permissions. AFAICT, these are required for repositories where GitHub Actions cannot destroy (i.e. write) things. See https://github.com/github/codeql-action/issues/464 and search for "permissions" in the latest CodeQL [documentation](https://github.com/github/codeql-action#usage).

```suggestion

    permissions:
      security-events: write

```